### PR TITLE
Class HasUILabel

### PR DESCRIPTION
**Pull request by fisx**
Produce string representations of aula types for the UI in a more uniform way.

### CODE REVIEW

**Comment by fisx** [comment]
Ready for review!

File: src/EventLog.hs [hunk 1]
```diff
@@ -143,13 +143,12 @@ instance CSV.ToRecord URLEventLogItem where
...
+                            <> showTrigger trigger
             , objLink topic
             ]
           where
-            p1 = phaseName fromPhase
-            p2 = phaseName toPhase
-
             showTrigger (PhaseTransitionTriggeredBy _)           = " (von Hand ausgelöst)"
             showTrigger PhaseTransitionTriggeredByTimeout        = " (Zeit ist abgelaufen)"
             showTrigger PhaseTransitionTriggeredByAllIdeasMarked = " (alle Ideen sind geprüft)"
```
Comment (np) [inline src/EventLog.hs]: Should `showTrigger` be a `labelS`

File: src/Frontend/Filter.hs [hunk 2]
```diff
@@ -180,19 +179,18 @@ instance Filter   SortUsersBy where
         byTime  = by $ createdAt . to Data.Ord.Down
         byName  = by userLogin
         byClass = by userSchoolClass
-        byRole  = by $ userRole . to (roleLabel :: Role -> ST)
+        byRole  = by $ userRole . labeledS
```
Comment (np) [inline src/Frontend/Filter.hs]: :+1:

File: src/Frontend/Page/Login.hs [hunk 3]
```diff
@@ -101,7 +101,7 @@ instance ToHtml LoginDemoHints where
                     th_ "password"
                 (\u -> tr_ $ do
                     td_ . toHtml $ u ^. userLogin . unUserLogin
-                    td_ . toHtml $ (roleLabel $ u ^. userRole :: ST)
+                    td_ . toHtml $ u ^. userRole . labeledST
```
Comment (np) [inline src/Frontend/Page/Login.hs]: Good candidates for the `html` getter.

File: src/Frontend/Page/Overview.hs [hunk 4]
```diff
@@ -141,7 +141,7 @@ instance ToHtml PageIdeasInDiscussion where
                                  , class_ "theme-grid-item-image"
                                  ]
                             div_ [class_ "theme-grid-item-text"] $ do
-                                span_ [class_ "theme-grid-item-phase"] . toHtml . phaseName
+                                span_ [class_ "theme-grid-item-phase"] . toHtml . labelST
                                     $ topic ^. topicPhase
```
Comment (np) [inline src/Frontend/Page/Overview.hs]: `topic ^. topicPhase . labeledST . html`

File: src/Types.hs [hunk 5]
```diff
@@ -1172,20 +1172,21 @@ instance HasUriPart IdeaSpace where
 instance HasUriPart SchoolClass where
     uriPart = fromString . showSchoolClass
 
+instance HasLabelS IdeaSpace where
+    labelS = \case
+        SchoolSpace    -> "Schule"
+        (ClassSpace c) -> labelS c
+
+instance HasLabelS SchoolClass where
+    labelS = fromString . cs . view className
```
Comment (np) [inline src/Types.hs]: This one is borderline but I agree we want it.

**Comment by np** [comment]
I made some small comments.

I prefer it without 1e8a500 "Renmae HasLabelS => HasUILabel."

**Comment by fisx** [comment]
@np you got outvoted on reverting 1e8a500 (rationale: "ui" says that it yields something the user is exposed to; it gets translated and everything; "string" and "label" are more general and vague).

I made all your other changes.  Thanks for the review!

**Comment by fisx** [comment]
rebased (due to the hick-up on master and a few spourious commits in this PR)

File: src/Types.hs [hunk 6]
```diff
@@ -476,16 +477,16 @@ data Phase =
...
-    PhaseRefinement{} -> "Ausarbeitungsphase"
-    PhaseRefFrozen{}  -> "Ausarbeitungsphase"
-    PhaseJury         -> "Prüfungsphase"
-    PhaseVoting{}     -> "Abstimmungsphase"
-    PhaseVotFrozen{}  -> "Abstimmungsphase"
-    PhaseResult       -> "Ergebnisphase"
+instance HasUILabel Phase where
+    uilabel = \case
+        PhaseWildIdea     -> "Wilde-Ideen-Phase"  -- FIXME: or is that unreachable code?
+        PhaseWildFrozen   -> "Wilde-Ideen-Phase"  -- FIXME: or is that unreachable code?
```
Comment (Mikolaj) [inline src/Types.hs]: As I said, this is currently unreachable, but may be in the future and currently tests reach it.

File: src/Types.hs [hunk 7]
```diff
@@ -1157,20 +1172,21 @@ instance HasUriPart IdeaSpace where
 instance HasUriPart SchoolClass where
     uriPart = fromString . showSchoolClass
 
+instance HasUILabel IdeaSpace where
+    uilabel = \case
+        SchoolSpace    -> "Schule"
+        (ClassSpace c) -> uilabel c
```
Comment (Mikolaj) [inline src/Types.hs]: It's done differently here:

```
src/Frontend/Page/Overview.hs-        h SchoolSpace = "Schule"
src/Frontend/Page/Overview.hs:        h (ClassSpace c) = "Klasse " <> c ^. className . html
```

and in some other places.

**Comment by Mikolaj** [comment]
Apart of the `ClassSpace` issue, LGTM.

File: src/Types.hs (hunk 6 above)
Comment (fisx) [inline src/Types.hs]: Sorry, outdated FIXME.  (Anyway if our future selves forget this, they will just take it out and run into the test case where it's used.)

**Comment by fisx** [comment]
Good catch @Mikolaj, thanks!  I hope I got them all now...

**Pull request merged**